src/DataStructures/Tensor/Expressions/Magnitude.hpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions euclidean_magnitude and magnitude
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+/*!
+ * \ingroup Tensor
+ * \brief compute the Euclidean magnitude of a rank-1 tensor
+ *
+ * \returns the Euclidean magnitude of the rank-1 tensor
+ *
+ * \details
+ * Computes the square root of the sum of the squares of the components of
+ * the rank-1 tensor.
+ *
+ */
+template <typename Index>
+DataVector euclidean_magnitude(
```

Comment:
My argument would be that if someone wants the magnitude of a vector w.r.t. the full metric and forget to provide the metric that's sort of on them. In short, I'm not sure how much hand holding we should do for this simple of a thing, but if others feel it adds a lot then I'm happy to have it stay as is